concordium-std/src/state_btree.rs

```diff
@@ -0,0 +1,2132 @@
+            self.key_value.remove(key);
+        }
+    }
+
+    /// Get a reference to the value corresponding to the key.
+    pub fn get(&self, key: &K) -> Option<StateRef<V>>
+    where
+        K: Serialize,
+        V: Serial + DeserialWithState<StateApi>, {
+        // Minor optimization in the case of the empty collection. Since the length is
+        // tracked by the ordered set, we can return early, saving a key lookup.
+        if self.key_order.is_empty() {
+            None
+        } else {
+            self.key_value.get(key)
+        }
+    }
+
+    /// Get a mutable reference to the value corresponding to the key.
+    pub fn get_mut(&mut self, key: &K) -> Option<StateRefMut<V, StateApi>>
+    where
+        K: Serialize,
+        V: Serial + DeserialWithState<StateApi>, {
+        // Minor optimization in the case of the empty collection. Since the length is
+        // tracked by the ordered set, we can return early, saving a key lookup.
+        if self.key_order.is_empty() {
+            None
+        } else {
+            self.key_value.get_mut(key)
+        }
+    }
+
+    /// Returns `true` if the map contains a value for the specified key.
+    pub fn contains_key(&self, key: &K) -> bool
+    where
+        K: Serialize + Ord, {
+        self.key_order.contains(key)
+    }
+
+    /// Returns the smallest key in the map, which is strictly larger than the
+    /// provided key. `None` meaning no such key is present in the map.
+    pub fn higher(&self, key: &K) -> Option<StateRef<K>>
+    where
+        K: Serialize + Ord, {
+        self.key_order.higher(key)
+    }
+
+    /// Returns the smallest key in the map, which is equal or larger than the
+    /// provided key. `None` meaning no such key is present in the map.
+    pub fn eq_or_higher(&self, key: &K) -> Option<StateRef<K>>
+    where
+        K: Serialize + Ord, {
+        self.key_order.eq_or_higher(key)
+    }
+
+    /// Returns the largest key in the map, which is strictly smaller than the
+    /// provided key. `None` meaning no such key is present in the map.
+    pub fn lower(&self, key: &K) -> Option<StateRef<K>>
+    where
+        K: Serialize + Ord, {
+        self.key_order.lower(key)
+    }
+
+    /// Returns the largest key in the map, which is equal or smaller than the
+    /// provided key. `None` meaning no such key is present in the map.
+    pub fn eq_or_lower(&self, key: &K) -> Option<StateRef<K>>
+    where
+        K: Serialize + Ord, {
+        self.key_order.eq_or_lower(key)
+    }
+
+    /// Returns a reference to the first key in the map, if any. This key is
+    /// always the minimum of all keys in the map.
+    pub fn first_key(&self) -> Option<StateRef<K>>
+    where
+        K: Serialize + Ord, {
+        self.key_order.first()
+    }
+
+    /// Returns a reference to the last key in the map, if any. This key is
+    /// always the maximum of all keys in the map.
+    pub fn last_key(&self) -> Option<StateRef<K>>
+    where
```

Comment:
Since all of these just forward calls it'd probably be useful if they are inline(always). That probably helps to inline in case people compile with `-z` or `-s` options which is typical.